pkg/controller/controller_integration_test.go

```diff
@@ -41,62 +43,60 @@ func TestIntegrationKangalController(t *testing.T) {
-			LabelSelector: "app=loadtest-master",
-		})
-
-		watchEvent, err := waitfor.Resource(watchObj, (waitfor.Condition{}).PodRunning)
-		require.NoError(t, err)
-
-		pod := watchEvent.Object.(*coreV1.Pod)
-		assert.Equal(t, coreV1.PodRunning, pod.Status.Phase)
+	// Checking master pod is created
+	watchObj, _ = client.CoreV1().Pods(expectedLoadtestName).Watch(context.Background(), metaV1.ListOptions{
+		LabelSelector: "app=loadtest-master",
 	})
+	watchEvent, err = waitfor.Resource(watchObj, (waitfor.Condition{}).PodRunning)
+	require.NoError(t, err)
+	pod := watchEvent.Object.(*coreV1.Pod)
+	require.Equal(t, coreV1.PodRunning, pod.Status.Phase)
 
-	t.Run("Checking loadtest is in Running state", func(t *testing.T) {
-		watchObj, _ := clientSet.KangalV1().LoadTests().Watch(context.Background(), metaV1.ListOptions{
-			FieldSelector: fmt.Sprintf("metadata.name=%s", expectedLoadtestName),
-		})
-
-		watchEvent, err := waitfor.Resource(watchObj, (waitfor.Condition{}).LoadTestRunning)
-		require.NoError(t, err)
-
-		loadtest := watchEvent.Object.(*loadTestV1.LoadTest)
-		assert.Equal(t, loadTestV1.LoadTestRunning, loadtest.Status.Phase)
+	// Checking loadtest is in Running state
+	watchObj, _ = clientSet.KangalV1().LoadTests().Watch(context.Background(), metaV1.ListOptions{
+		FieldSelector: fmt.Sprintf("metadata.name=%s", expectedLoadtestName),
 	})
+	watchEvent, err = waitfor.Resource(watchObj, (waitfor.Condition{}).LoadTestRunning)
+	require.NoError(t, err)
+	loadtest := watchEvent.Object.(*loadTestV1.LoadTest)
+	require.Equal(t, loadTestV1.LoadTestRunning, loadtest.Status.Phase)
 
-	t.Run("Checking loadtest is in Finished state", func(t *testing.T) {
-		// We do run fake provider which has 10 sec sleep before finishing job
-		watchObj, _ := clientSet.KangalV1().LoadTests().Watch(context.Background(), metaV1.ListOptions{
-			FieldSelector: fmt.Sprintf("metadata.name=%s", expectedLoadtestName),
-		})
-
-		watchEvent, err := waitfor.Resource(watchObj, (waitfor.Condition{}).LoadTestFinished)
-		require.NoError(t, err)
-
-		loadtest := watchEvent.Object.(*loadTestV1.LoadTest)
-		assert.Equal(t, loadTestV1.LoadTestFinished, loadtest.Status.Phase)
+	// Checking loadtest is in Finished state
+	// We use loadtests with fake provider which only runs for 10 sec and then finishes the job
+	watchObj, _ = clientSet.KangalV1().LoadTests().Watch(context.Background(), metaV1.ListOptions{
+		FieldSelector: fmt.Sprintf("metadata.name=%s", expectedLoadtestName),
 	})
+	watchEvent, err = waitfor.Resource(watchObj, (waitfor.Condition{}).LoadTestFinished)
+	require.NoError(t, err)
+	loadtest = watchEvent.Object.(*loadTestV1.LoadTest)
+	require.Equal(t, loadTestV1.LoadTestFinished, loadtest.Status.Phase)
+
+	// Checking finished loadtest is deleted
+	// SyncHandler runs every 10s for integration test.
+	// We expect SyncHandler to delete finished loadtest after 10s but wait 30s to be sure
+	time.Sleep(30 * time.Second)
```

Comment:
Just to slightly speed up the test - probably it makes sense to run this check in the loop 3 times with 10s interval, or 6 times with 5s and fail only when after several times the test still persists